src/components/Checkbox.tsx

```diff
@@ -75,21 +22,64 @@ const Checkbox: React.FC<IProps> = ({
       callback()
     }
   }
+
+  const checkboxContainerAttributes = {
+    display: "inline-block",
+    verticalAlign: "middle",
+    className: className,
+    onClick: handleClick,
+  }
+
+  const hiddenCheckboxAttributes = {
+    type: "checkbox",
+    checked: checked,
+    readOnly: true,
+    ...rest,
+  }
+
+  const styledCheckboxAttributes = {
+    "aria-hidden": true,
+    className: "styled-checkbox",
+    display: "inline-block",
+    w: `${size}rem`,
+    h: `${size}rem`,
+    minW: `${size}rem`,
+    bg: `${checked ? "primary400" : "background"}`,
+    border: "1px",
+    borderStyle: "solid",
+    borderColor: "black50",
+    borderRadius: "3px",
+    transition: "all 150ms",
+    _hover: {
+      boxShadow: "tableItemBoxShadow",
+      border: "1px",
+      borderStyle: "solid",
+      borderColor: "primary600",
+      transition: "transform 0.1s",
+      transform: "scale(1.02)",
+    },
+  }
+
   return (
-    <CheckboxContainer className={className} onClick={handleClick}>
-      <HiddenCheckbox type="checkbox" checked={checked} readOnly {...rest} />
-      <StyledCheckbox
-        aria-hidden="true"
-        checked={checked}
-        className="styled-checkbox"
-        size={size}
-      >
-        <Icon checked={checked} viewBox="0 0 24 24">
+    <Box {...checkboxContainerAttributes}>
+      <VisuallyHiddenInput {...hiddenCheckboxAttributes} />
+      <Box {...styledCheckboxAttributes}>
+        <svg
```

Comment:
Wondering if we could use the Chakra `Icon` component in here instead of the <svg>. Check [this](https://chakra-ui.com/docs/components/icon/usage#using-the-icon-component) for reference.

Reply:
I struggled with the Chakra Icon a few hours but it turned out that component comes with an amount of CSS that I couldn't figure out how to override properly... The svg was moved a little bit lower and polyline even lower.
This was the result regardless of my attempts: 
![Screenshot 2022-09-13 at 22 17 12](https://user-images.githubusercontent.com/63505124/190000841-9c4f1772-48dd-4339-9cec-d3ebc0c3ab33.png)

I propose to keep temporarily the `<svg>` until the next iteration with the Chakra Checkbox. Or if you have any suggestions on how to use the Chakra Icon I'll try again.